tools/transform.cpp

```diff
@@ -370,11 +370,11 @@ static void calculateAndInitConstants(Transform &t) {
   unsigned num_ptrinputs = 0;
   const auto &inputs = t.src.getInputs();
   for (auto &arg : inputs) {
-    // An argument with aggregate type with pointer member isn't regarded
-    // as a source of non-local pointer, because its member should be extracted
-    // with extractelement / extractvalue instructions, which are also counted
-    // as source of non-local blocks.
+    // An argument with aggregate type with pointer member is dealt separately
     num_ptrinputs += arg.getType().isPtrType();
+
+    if (auto aty = arg.getType().getAsAggregateType())
+      num_ptrinputs += aty->numPointerElements();
```

Comment:
What's the reason for this change?  Can you give an example pls?

Reply:
I saw now your example in the other page. Can we handle that separately so we try to be less conservative. It's important to keep the number of blocks very low.

Reply:
Personally, I prefer leaving the conservative version here (so it addresses correctness first) and making another patch that optimizes it. It helps the added unit tests work as well. What do you think?